Homework4

WojtekMs opened the pull request.

Marcinati committed:
double > (#78)

ziobron committed:
Merge branch stl_module3 (#95)

* Add module3 for STL

* Update algorytmy.md

* Delete index.html

* Update ZlozonoscObliczeniowa.md

* Update Iterator.md

* Stl module3 - Utworzenie prezentacji (#94)

* workflows

* fixed typos and added index.html

* added reveals.js comments and fixed some more typos

* Apply suggestions from code review

Co-authored-by: Penthurion <68010070+Penthurion@users.noreply.github.com>
Co-authored-by: WBigaj <wiktorbigaj@wp.pl>

* changed 'math' to 'tex'

* added iframe for comparison

* some fixes

* new line added

* Apply suggestions from code review

* math equations now display correctly

Co-authored-by: Penthurion <68010070+Penthurion@users.noreply.github.com>
Co-authored-by: WBigaj <wiktorbigaj@wp.pl>
Co-authored-by: Łukasz Ziobroń <lukasz@coders.school>

* Unwanted stuff removal :)

Co-authored-by: Mateusz Adamski <nauka.programowania.ma@gmail.com>
Co-authored-by: nauka-programowania-MA <60320742+nauka-programowania-MA@users.noreply.github.com>
Co-authored-by: MrMijagi <45121219+MrMijagi@users.noreply.github.com>
Co-authored-by: Penthurion <68010070+Penthurion@users.noreply.github.com>
Co-authored-by: WBigaj <wiktorbigaj@wp.pl>

dawidbil committed:
generated pdf (#96)

* added pdf, fixed one slide

* removed ($) dollar signs which caused trouble

* generated fixed pdf

ziobron committed:
Stl module4 (#97)

* add module4 for STL

* Update istream_fstream.md

* Stl module4 - Utworzenie prezentacji (#93)

* workflows/module4.yml

* fixed typos

* added index.html and reveal.js comments

* Apply suggestions from code review

Co-authored-by: Penthurion <68010070+Penthurion@users.noreply.github.com>

* ku -> iteratora do przodu

* peek() line fixed

* added spans

* Some fixes

* added fixed from code review

* Delete main.cpp

Co-authored-by: Penthurion <68010070+Penthurion@users.noreply.github.com>
Co-authored-by: Łukasz Ziobroń <lukasz@coders.school>

Co-authored-by: Mateusz Adamski <nauka.programowania.ma@gmail.com>
Co-authored-by: nauka-programowania-MA <60320742+nauka-programowania-MA@users.noreply.github.com>
Co-authored-by: MrMijagi <45121219+MrMijagi@users.noreply.github.com>
Co-authored-by: Penthurion <68010070+Penthurion@users.noreply.github.com>

arafalski committed:
Fixed '!' misplacement in complexity presentation (#98)

* Fixed mistakes in complexity presentation

* Revert "Fixed mistakes in complexity presentation"

This reverts commit c3e0a348c4ef869af4a83a73baf8fdf63931626c.

* Fixed mistakes in complexity presentation

dawidbil committed:
Normalizacja i ujednolicanie (#100)

* added generated pdf, fixed agenda in index.html

* fixed image in pdf

* normalizacja i ujednolicanie

Marcinati committed:
missing "c" in "Hierarchia" and redundant brackets (#101)

* missing "c" in "Hierarchia"

* redundant brackets

arafalski committed:
Fix for tests in advanced calculator (#99)

* Bugfix for calculator tests

* Updated error codes

ziobron committed:
pasta o iteratorach -> pasta o strumieniach

Commit:
fixed a typo in test.cpp (#102)

* fixed a typo in test.cpp

* Update module3/homework/advanced_calculator/test.cpp

Co-authored-by: Kuba C <jakub.cieplinski@outlook.com>
Co-authored-by: Łukasz Ziobroń <lukasz@coders.school>

nauka-programowania-MA committed:
Fix tests

ziobron committed:
Points for homeworks added

WojtekMs committed:
configuring task

WojtekMs committed:
FormatIngredients definition added

WojtekMs committed:
FormatIngredients fix

WojtekMs committed:
FormatRecipit added

WojtekMs committed:
formatIngredients fix

WojtekMs committed:
formatRecipe fix

WojtekMs committed:
file opening mode changed to app

WojtekMs committed:
adding adaptors folder

WojtekMs committed:
Update README.md

WojtekMs committed:
Update README.md

WojtekMs committed:
Stack structure notes

WojtekMs committed:
Update README.md

WojtekMs committed:
Update README.md

WojtekMs committed:
Update README.md

WojtekMs committed:
Update README.md

WojtekMs committed:
Update README.md

WojtekMs committed:
Update README.md

WojtekMs committed:
Queue & Priority Queue added

WojtekMs committed:
Update README.md

MaciejSalamonski commented on module4/homework/appendNewRecipe/AppendNewRecipe.cpp:
```diff
@@ -0,0 +1,62 @@
+                                const std::list<std::string>& ingredients,
+                                const std::deque<std::pair<size_t, char>>& amount) {
+    std::stringstream stream;
+    stream << "Skladniki:\n";
```
I suggest to make these sentences (Skladniki, Kroki, etc) as const or constexpr variables.

WBigaj replied:
+1

nauka-programowania-MA replied:
+1

MaciejSalamonski reviewed (approved):
Hi nice code!

J-Cieplinski committed:
changed test name to be more appropriate (#106)

* changed test name to be more appropriate

* Fixed one omitted naming of root

Co-authored-by: Jakub Cieplinski <PierwszyKrytykRP@users.noreply.github.com>

J-Cieplinski committed:
Fixed Factorial test, changed naming of one test (#112)

[19:08] MateuszAdamski: Z tymi ujemnymi silniami to co innego miałem na myśli co innego zapisałem :slight_smile: chodziło mi, że liczy -(num!). Innymi słowy -5! to negate(5!). Testy pisałem parę dni później i pewnie nie zauważyłem jakie głupoty piszę :wink:

Co-authored-by: Jakub Cieplinski <PierwszyKrytykRP@users.noreply.github.com>

WBigaj commented on module4/homework/appendNewRecipe/AppendNewRecipe.cpp:
```diff
@@ -0,0 +1,62 @@
+bool AppendNewRecipe(std::vector<std::string> steps,
+                     const std::list<std::string>& ingredients,
+                     const std::deque<std::pair<size_t, char>>& amount) {
+    std::ofstream file("recipes.txt", std::ios::app);
```
I think file path should be const variable.

nauka-programowania-MA replied:
+1

WBigaj reviewed (commented):
Why have you PRed some many files? I have checked your code and left my feedback.

nauka-programowania-MA commented on module4/homework/appendNewRecipe/AppendNewRecipe.cpp:
```diff
@@ -0,0 +1,62 @@
+#include <algorithm>
+#include <fstream>
+
+bool AppendNewRecipe(std::vector<std::string> steps,
```
don;t copy vector

WojtekMs replied:
I copied the function signature from the STL#4 lesson zadanie 1. I will correct it, however it should be also fixed on the website.

nauka-programowania-MA commented on module4/homework/appendNewRecipe/AppendNewRecipe.cpp:
```diff
@@ -0,0 +1,62 @@
+                     const std::deque<std::pair<size_t, char>>& amount) {
+    std::ofstream file("recipes.txt", std::ios::app);
+    if (file.is_open()) {
+        auto text = FormatRecipit(steps, ingredients, amount).str();
```
const

nauka-programowania-MA commented on module4/homework/appendNewRecipe/AppendNewRecipe.cpp:
```diff
@@ -0,0 +1,62 @@
+                   begin(amount),
+                   begin(formattedIngredients),
+                   [](const auto& ingredient, const auto& pair) {
+                       auto [amount, unit] = pair;
```
const

nauka-programowania-MA commented on module4/homework/appendNewRecipe/AppendNewRecipe.cpp:
```diff
@@ -0,0 +1,62 @@
+                       }
+                       if (unit == 'm') {
+                           formattedUnit = "mililitrow";
+                       }
```
switch (char) {

}

nauka-programowania-MA commented on module4/homework/appendNewRecipe/AppendNewRecipe.cpp:
```diff
@@ -0,0 +1,62 @@
+                       if (unit == 'm') {
+                           formattedUnit = "mililitrow";
+                       }
+                       return std::to_string(amount) + ' ' + formattedUnit + ' ' + ingredient;
```
Use sstringstream to avoid unnecessary creating of temporary string object.

WojtekMs replied:
True, thanks for this recommendation :)

nauka-programowania-MA commented on module4/homework/appendNewRecipe/AppendNewRecipe.cpp:
```diff
@@ -0,0 +1,62 @@
+    return formattedIngredients;
+}
+
+std::stringstream FormatRecipit(std::vector<std::string> steps,
```
don't copy vector

nauka-programowania-MA commented on module4/homework/appendNewRecipe/AppendNewRecipe.cpp:
```diff
@@ -0,0 +1,62 @@
+    std::stringstream stream;
+    stream << "Skladniki:\n";
+    auto formattedIngredients = FormatIngredients(ingredients, amount);
+    for (auto line : formattedIngredients) {
```
const auto&

nauka-programowania-MA commented on module4/homework/appendNewRecipe/AppendNewRecipe.cpp:
```diff
@@ -0,0 +1,62 @@
+    for (size_t i = 0; i < steps.size(); ++i) {
+        stream << i + 1 << ") " << steps[i] << ".\n";
+    }
+    stream << "___________________________________\n";
```
Create std::string(occurrences, char)

WojtekMs replied:
I agree with you that it is much easier to create a dash like this, however:
In the tests you created this variable:
```cpp
constexpr const char kExpected[] =
    "Skladniki:\n20 gram cukru,\n1 szklanka(i) maki,"
    "\n40 mililitrow rumu,\n\nKroki:\n1) Wsypac do miski"
    " 20 gram cukru.\n2) Dorzucic 1 szklanke maki.\n3)"
    " Dokladnie wymieszac.\n4) Nalac 40ml rumu do kieliszka.\n"
    "5) Wypic kieliszek.\n6) Wysypac zawartosc miski."
    "\n___________________________________\n";
```
My output has to be equal to this text, so I just copied the dash, I was too lazy to count the number of dashes you put in kExpected.

WBigaj committed:
Fixed basic check in ChangePos + minor formatting fixes for consistency (#126)

WojtekMs committed:
Merge branch 'master' of github.com:coders-school/stl into homework4

WojtekMs committed:
adding const & to steps vector

WojtekMs committed:
AppendNewRecipe function fixed

WojtekMs committed:
FormatIngredients function added

WojtekMs committed:
FormatRecipit function fixed

WojtekMs committed:
using sstream to avoid creating temp string objects
